db/migrate/20140219183343_migrate_permissions.rb

```diff
@@ -22,34 +22,34 @@ def resource_type
     @resource_type ||= permissions.first.try(:resource_type)
   end
 
-  taxonomy_join_table = "taxable_taxonomies"
+  taxonomy_join_table = :taxable_taxonomies
   has_many taxonomy_join_table, :dependent => :destroy, :as => :taxable, :foreign_key => 'taxable_id'
-  has_many :locations,     :through => taxonomy_join_table, :source => :taxonomy,
-           :conditions => "taxonomies.type='Location'", :validate => false
-  has_many :organizations, :through => taxonomy_join_table, :source => :taxonomy,
-           :conditions => "taxonomies.type='Organization'", :validate => false
+  has_many :locations,     lambda{where("taxonomies.type='Location'")}, :through => taxonomy_join_table, :source => :taxonomy,
+           :validate => false
+  has_many :organizations, lambda{where("taxonomies.type='Organization'")}, :through => taxonomy_join_table, :source => :taxonomy,
```

Comment:
Still failing due to this, doesn't look Rails 3 compatible to me, everything seems to point to using :conditions still, but perhaps with a Proc instead.